tutorials/performance/pipeline_compilations.rst

```diff
@@ -0,0 +1,77 @@
+
+Pipeline Pre-Compilation Monitors
+---------------------------------
+
+Compiling pipelines ahead of time is the main mechanism Godot uses to mitigate shader stutters, but it's not a perfect solution. Being aware of the situations that can lead to pipeline stutters can be very helpful, and the workarounds are pretty straightforward compared to previous versions. These workarounds may be less necessary over time on future versions of Godot as more detection techniques are implemented.
+
+The Godot debugger offers monitors for tracking the amount of pipelines created by the game and the step that triggered their compilation. You can keep an eye on these monitors as the game runs to identify potential sources of shader stutters without having to wipe your driver cache every time you wish to test. Sudden increases of these values outside of loading screens can show up as hitches during gameplay the first time someone plays the game on their system. **It is recommended you take a look at these monitors to identify possible sources of stutter for your players**, as you might be unable to experience them yourself without deleting your driver cache or testing on a weaker system.
+
+.. figure:: img/pipeline_compilations_monitors.png
+   :align: center
+   :alt: Screenshot of the Godot Pipeline Compilations monitors
+
+   Pipeline compilations of one of the demo projects.
+
+.. note:: We can see the pipelines compiled during gameplay and
+          verify which steps could possibly cause stuttters. Note
+          that these values will only increase and never go down,
+          as deleted pipelines are not tracked by these monitors
+          and pipelines may be erased and recreated during gameplay.
+
+- **Canvas**: Compiled when drawing a 2D node. The engine does not currently feature pre-compilation for 2D elements and the stutters will show up when the node is drawn for the first time.
+- **Mesh**: Compiled as part of loading a 3D mesh and identifying what pipelines can be pre-compiled from its properties. These can lead to stutters if a mesh is loaded from the game loop, but they can be mitigated if the mesh is loaded by using a background thread. **Modifiers that are part of nodes such as material overrides can't be compiled on this step**. 
+- **Surface**: Compiled when a frame is about to be drawn and 3D objects were instanced on the scene tree for the first time. This can also include compilation for nodes that aren't even visible on the scene tree. The stutter will occur only on the first frame the node is added to the scene, which won't result in an obvious stutter if it happens right after a loading screen.
```

Comment:
```suggestion
- **Surface**: Compiled when a frame is about to be drawn and 3D objects were instanced on the scene tree for the first time. This can also include compilation for nodes that aren't even visible on the scene tree. The stutter will occur only on the first frame the node is added to the scene, which won't result in an obvious stutter if it happens right after a loading screen. Commonly, compilations happen here as a result of using material overrides.
```

Reply:
I feel this one may be a bit misleading as currently Surface compilations are a pretty big portion due to reasons such as just not being covered by the use ArrayMesh. We probably don't want to lead users down the path of taking out all overrides and not seeing that reflected in the monitor just yet.